src/main/scala/firrtl/ir/Serializer.scala

```diff
@@ -0,0 +1,270 @@
+    case UIntLiteral(value, width) =>
+      b ++= "UInt" ; s(width) ; b ++= "(\"h" ; b ++= value.toString(16) ; b ++= "\")"
+    case SIntLiteral(value, width) =>
+      b ++= "SInt" ; s(width) ; b ++= "(\"h" ; b ++= value.toString(16) ; b ++= "\")"
+    case FixedLiteral(value, width, point) =>
+      b ++= "Fixed"
+      if(width != UnknownWidth) { b += '<' ; s(width) ; b += '>' }
+      s(point) ; b ++= "(\"h" ; b ++= value.toString(16) ; b ++= "\")"
+    case DoPrim(op, args, consts, _) =>
+      b ++= op.toString ; b += '(' ; s(args, ", ", consts.isEmpty) ; s(consts, ", ") ; b += ')'
+    // WIR
+    case firrtl.WVoid => b ++= "VOID"
+    case firrtl.WInvalid => b ++= "INVALID"
+    case firrtl.EmptyExpression => b ++= "EMPTY"
+  }
+
+  //scalastyle:off cyclomatic.complexity method.length
+  private def s(node: Statement)(implicit b: StringBuilder, indent: Int): Unit = node match {
+    case DefWire(info, name, tpe) => b ++= "wire " ; b ++= name ; b ++= " : " ; s(tpe) ; s(info)
+    case DefRegister(info, name, tpe, clock, reset, init) =>
+      b ++= "reg " ; b ++= name ; b ++= " : " ; s(tpe) ; b ++= ", " ; s(clock) ; b ++= " with :" ; newLineAndIndent(1)
+      b ++= "reset => (" ; s(reset) ; b ++= ", " ; s(init) ; b += ')' ; s(info)
+    case DefInstance(info, name, module, _) => b ++= "inst " ; b ++= name ; b ++= " of " ; b ++= module ; s(info)
+    case DefMemory(info, name, dataType, depth, writeLatency, readLatency, readers, writers,
+    readwriters, readUnderWrite) =>
+      b ++= "mem " ; b ++= name ; b ++= " :" ; s(info) ; newLineAndIndent(1)
+      b ++= "data-type => "     ; s(dataType) ; newLineAndIndent(1)
+      b ++= "depth => "         ; b ++= depth.toString() ; newLineAndIndent(1)
+      b ++= "read-latency => "  ; b ++= readLatency.toString ; newLineAndIndent(1)
+      b ++= "write-latency => " ; b ++= writeLatency.toString ; newLineAndIndent(1)
+      readers.foreach{ r => b ++= "reader => " ; b ++= r ; newLineAndIndent(1) }
+      writers.foreach{ w => b ++= "writer => " ; b ++= w ; newLineAndIndent(1) }
+      readwriters.foreach{ r => b ++= "readwriter => " ; b ++= r ; newLineAndIndent(1) }
+      b ++= "read-under-write => " ; b ++= readUnderWrite.toString
+    case DefNode(info, name, value) => b ++= "node " ; b ++= name ; b ++= " = " ; s(value) ; s(info)
+    case Conditionally(info, pred, conseq, alt) =>
+      b ++= "when " ; s(pred) ; b ++= " :" ; s(info)
+      newLineAndIndent(1) ; s(conseq)(b, indent + 1)
+      if(alt != EmptyStmt) {
+        newLineAndIndent() ; b ++= "else :"
+        newLineAndIndent(1) ; s(alt)(b, indent + 1)
+      }
+    case Block(stmts) =>
+      val it = stmts.iterator
+      while(it.hasNext) {
+        s(it.next)
+        if(it.hasNext) newLineAndIndent()
+      }
+    case PartialConnect(info, loc, expr) => s(loc) ; b ++= " <- " ; s(expr) ; s(info)
+    case Connect(info, loc, expr) => s(loc) ; b ++= " <= " ; s(expr) ; s(info)
```

Comment:
I would also put `Connect` high. Every `Wire` and `Reg` should have at least 1 `Connect`